crates/http/src/backend/hyper/handler.rs

```diff
@@ -0,0 +1,81 @@
+use hyper_util::rt::{TokioExecutor, TokioIo, TokioTimer};
+use hyper_util::server::conn::auto;
+use scuffle_context::ContextFutExt;
+use tokio::io::{AsyncRead, AsyncWrite};
+
+use crate::error::Error;
+use crate::service::{HttpService, HttpServiceFactory};
+
+/// Helper function used by hyper server to handle incoming connections.
+pub async fn handle_connection<F, S, I>(
+    ctx: scuffle_context::Context,
+    service: S,
+    io: I,
+    #[cfg(feature = "http1")] http1: bool,
+    #[cfg(feature = "http2")] http2: bool,
```

Comment:
We should not change the function signature depending on if these are enabled or not I think. Instead we should just accept both and do nothing if the feature is off?